version: '3.8'

services:
  audit-service:
    build: .
    command: python /app/main.py
    depends_on:
      - user-management-service
      - manager
    volumes:
      - ./app:/app
    ports:
      - ${AUDIT_SERVICE_PORT}:${AUDIT_SERVICE_PORT}
    env_file:
      - .env

  user-management-service:
    image: ghcr.io/projekt-datascience/backend-user-management:release
    depends_on:
      - manager
    ports:
      - ${USER_MANAGEMENT_SERVICE_PORT}:${USER_MANAGEMENT_SERVICE_PORT}
    env_file:
      - .env

  db:
    image: postgres:latest
    restart: always
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
      PGPORT: ${DB_PORT}
    ports:
      - ${DB_PORT}:${DB_PORT}
    networks:
      - microservice-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 10s
      timeout: 5s
      retries: 5

  manager:
    image: ghcr.io/projekt-datascience/manager:release
    env_file:
      - .env